cpp/test/distance/sparse_l2_nn.cu

```diff
@@ -0,0 +1,494 @@
+/*
```

Comment:
T[his is a new file so it should only have the current year.

Reply:
What is the definition of 'publication' here? I have updated the files to add copyright in the year 2023 when I have edited them. The files where originally published (as draft PR) in 2022.

Reply:
The year should reflect the time the files become a part of the release branch, not the time they were created as a draft.